core/src/worker/workflow/machines/workflow_machines.rs

```diff
@@ -426,7 +426,10 @@ impl WorkflowMachines {
         // Update observed patches with any that were used in the task
         if let Some(next_complete) = self
             .last_history_from_server
-            .peek_next_wft_completed(self.last_processed_event)
+            // Our last_processed event is always a WorkflowTaskStarted, we don't care about
+            // internal flags in the current workflow task since we already know about those.
+            // Peek past the WorkflowTaskCompleted event for the current workflow task.
+            .peek_next_wft_completed(self.last_processed_event + 2)
```

Comment:
I don't love adding 2 explicitly here. I think it'd be better to add a bool argument like `skip_next` or something (hard to come up with a good name there). And then encode logic into the peek function directly to skip past the next immediate completed and look for the next one after that. I think it's just a slightly safer / more intentional approach.

Reply:
I agree, there are probably some edge cases that it doesn't cover like when the current WFT failed.
Need a better solution to ensure we look ahead past the "previous" (or current depending on perspective) task.